src/partition/ReceivedPartition.cpp

```diff
@@ -93,6 +94,28 @@ void ReceivedPartition::compute()
   // handle coupling mode first (i.e. serial participant)
   if (!utils::MasterSlave::isParallel()) { //coupling mode
     PRECICE_DEBUG("Handle partition data structures for serial participant");
+
+    if (_allowDirectAccess) {
+      // Prepare the bounding boxes
+      prepareBoundingBox();
+      // Filter out vertices not laying in the bounding box
+      mesh::Mesh filteredMesh("FilteredMesh", _dimensions, mesh::Mesh::MESH_ID_UNDEFINED);
+      // To discuss: maybe check this somewhere in the SolverInterfaceImpl, as we have now a similar check for the parallel case
+      PRECICE_CHECK(!_bb.empty(), "You are running in serial mode and the bounding box on mesh \"{}\", is empty. Did you call setBoundingBox with valid data?", _mesh->getName());
```

Comment:
That is a conceptual question. Since I am filtering the mesh according to the local bounding box, continuation does not make any sense without a defined bounding box. So, aborting would be the way to go here. I would keep the required bounding box filtering and definition for consistency reasons. If someone really wants to use preCICE in serial mode only and does not want to compute or provide any bounding box he can set a 'big enough' bounding box.
However, I am open for discussions here: I can also remove the filter here completely.

Reply:
I guess we can't say yet what would be better. Should we keep this open for the moment?
I have a tendency towards allowing no bounding box, but I have no good arguments yet :grin: 
Implementation-wise, the current solution is simpler, right?

Reply:
Simpler in the sense of consistent between serial and parallel execution. 

> I have a tendency towards allowing no bounding box

We should not require the user to change the code path when the same code is executed in serial or in parallel. This makes things just complicated. 'Allowing no bounding box' in serial and 'require' one in parallel would be hard to understand and not very user-friendly.

Reply:
Let's keep it as it is for the moment.